src/Internal/Service/Blockfrost.purs

```diff
@@ -116,3 +151,105 @@ handleBlockfrostResponse (Right { status: Affjax.StatusCode statusCode, body })
       body # lmap (ClientDecodeJsonError body)
         <<< (decodeAeson <=< parseJsonStringToAeson)
 
+handle404AsNothing
+  :: forall x. Either ClientError (Maybe x) -> Either ClientError (Maybe x)
+handle404AsNothing (Left (ClientHttpResponseError (Affjax.StatusCode 404) _)) =
+  Right Nothing
+handle404AsNothing x = x
+
+getDatumByHash
+  :: DataHash -> BlockfrostServiceM (Either ClientError (Maybe Datum))
+getDatumByHash dataHash = do
+  response <- blockfrostGetRequest (GetDatumCbor dataHash)
+  pure $ handle404AsNothing $ unwrapBlockfrostDatum <$> handleBlockfrostResponse
+    response
+
+getScriptInfo
+  :: ScriptHash
+  -> BlockfrostServiceM (Either ClientError (Maybe BlockfrostScriptInfo))
+getScriptInfo scriptHash = do
+  response <- blockfrostGetRequest (GetScriptInfo scriptHash)
+  pure $ handle404AsNothing $ handleBlockfrostResponse response
+
+getScriptByHash
+  :: ScriptHash -> BlockfrostServiceM (Either ClientError (Maybe ScriptRef))
+getScriptByHash scriptHash = runExceptT $ runMaybeT do
+  info <- MaybeT $ ExceptT $ getScriptInfo scriptHash
+  cbor <- MaybeT $ ExceptT $ do
+    response <- blockfrostGetRequest (GetScriptCbor scriptHash)
+    pure $ handle404AsNothing $ unwrapBlockfrostCbor <$>
+      handleBlockfrostResponse response
+  let
+    script = lmap (ClientDecodeJsonError "Error decoding script") $
+      wrapScriptType (scriptInfoType info) cbor
+  MaybeT $ ExceptT $ pure $ Just <$> script
+  where
+  wrapScriptType scriptType x = case scriptType of
+    NativeScript -> NativeScriptRef <$> decodeNativeScript x
```

Comment:
`/scripts/{script_hash}/cbor` endpoint works only for `plutus` scripts and always returns `null` for `native` ones. There is a separate endpoint which should be used for `native` scripts: https://docs.blockfrost.io/#tag/Cardano-Scripts/paths/~1scripts~1{script_hash}~1json/get.